cli/cli/src/main/java/org/projectnessie/nessie/cli/commands/RevertContentCommand.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.nessie.cli.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import jakarta.annotation.Nonnull;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.jline.utils.AttributedStringBuilder;
+import org.jline.utils.AttributedStyle;
+import org.projectnessie.client.api.CommitMultipleOperationsBuilder;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.CommitResponse;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.GetMultipleContentsResponse;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Reference;
+import org.projectnessie.nessie.cli.cli.BaseNessieCli;
+import org.projectnessie.nessie.cli.cmdspec.RevertContentCommandSpec;
+import org.projectnessie.nessie.cli.grammar.Node;
+import org.projectnessie.nessie.cli.grammar.Token;
+
+public class RevertContentCommand extends NessieCommittingCommand<RevertContentCommandSpec> {
+
+  @Override
+  protected CommitResponse executeCommitting(
+      @Nonnull BaseNessieCli cli,
+      RevertContentCommandSpec spec,
+      Branch branch,
+      CommitMultipleOperationsBuilder commit)
+      throws Exception {
+
+    List<ContentKey> contentKeys =
+        spec.getContentKeys().stream().map(ContentKey::fromPathString).collect(Collectors.toList());
+
+    String refName =
+        spec.getSourceRef() != null ? spec.getSourceRef() : cli.getCurrentReference().getName();
+    @SuppressWarnings("resource")
+    GetMultipleContentsResponse contents =
+        cli.mandatoryNessieApi()
+            .getContent()
+            .refName(refName)
+            .hashOnRef(spec.getSourceRefTimestampOrHash())
+            .keys(contentKeys)
+            .getWithResponse();
+
+    Map<ContentKey, Content> contentsMap = contents.toContentsMap();
+    Reference effectiveReference =
+        requireNonNull(
+            contents.getEffectiveReference(), "No effective reference, Nessie API v2 required");
+    for (ContentKey contentKey : contentKeys) {
+      Content content = contentsMap.get(contentKey);
+      if (content != null) {
+        commit.operation(Operation.Put.of(contentKey, content));
+      } else {
+        commit.operation(Operation.Delete.of(contentKey));
```

Comment:
I tend to think that deletes may be unexpected by the user of this command. Could we add a flag like `REVERT CONTENT OF ... [DELETE MISSING]`?